src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/SecurityServiceDispatcher.cs

```diff
@@ -0,0 +1,593 @@
+﻿using CoreWCF.Channels;
+using CoreWCF.Configuration;
+using CoreWCF.Description;
+using CoreWCF.Runtime;
+using CoreWCF.Security;
+using CoreWCF.Security.Tokens;
+using System;
+using System.Collections.Generic;
+using System.Security;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml;
+using System.Xml.Schema;
+
+namespace CoreWCF.Dispatcher
+{
+    public class SecurityServiceDispatcher : IServiceDispatcher
+    {
+        private IRequestReplyCorrelator _requestReplyCorrelator;
+        private BindingContext bindingContext;
+        TransportSecurityBindingElement transportSecurityBinding;
+        private SecurityProtocolFactory securityProtocolFactory;
+        private SecuritySessionServerSettings sessionServerSettings;
+        // Do we need it ?
+        SecurityListenerSettingsLifetimeManager settingsLifetimeManager;
+        private bool sessionMode;
+        private bool sendUnsecuredFaults;
+        private ChannelDispatcher securityAuthChannelDispatcher;
+        IChannel outerChannel;
+
+        public SecurityServiceDispatcher(TransportSecurityBindingElement transportSecurityBindingElement, BindingContext context, IServiceDispatcher serviceDispatcher)
+        {
```

Comment:
I wouldn't expect this to be scoped down to just TransportSecurityBindingElement. SecurityChannelListener takes a SecurityBindingElement. Restricting this would make it harder to add more features later.

Reply:
Can you rename the parameter too.